src/components/views/settings/AvatarSetting.js

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2019 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+
+import {ContextMenu, ContextMenuButton, toRightOf, useContextMenu} from "../../structures/ContextMenu";
+import sdk from "../../../index";
+
+// TODO: Why is rendering a box with an overlay so complicated? Can the DOM be reduced?
+const AvatarSetting = (props) => {
```

Comment:
ooi why a function with a bunch of bootstrap instead of a tiny PureComponent class?

Reply:
Ability to use `useContextMenu` hook simplifies things